### DISCUSSION TITLE
is there a way to properly rerender the component?

### DISCUSSION THREAD

**Discussion by iqbaltawakkal** [uploadcare/file-uploader#727] [blocks] [answered]
I'm building this vue component wrapper, the component set props for config and DOM property.
I notice that when props change reactively, file uploader is not rerender using latest props.
for example if you're changing the label using DOM localeDefinitionOverride.
is there a way to properly rerender? so I can watch the props and using rerender to render properly.

**Comment by nd0ut** [comment]
Hey @iqbaltawakkal,

The uploader will re-render on DOM property changes in most cases. However, there are some scenarios we haven't covered. Could you provide more detailed information?

1. Which l10n keys did you notice where the uploader doesn't re-render?
2. Are you updating the `localeDefinitionOverride` property by reference, or just updating in-place? In other words, are you actually calling the property setter or not?

**Comment by iqbaltawakkal** [comment]
Hi @nd0ut thanks for the response,
I use vue and doing something like this to dom property:
![image](https://github.com/user-attachments/assets/45fc5116-3954-41b6-aeb5-65a68de97262)
It does change only when first render, not reactively when props change (based on input user)
is this behaviour expected ?

**Reply by nd0ut** [reply]
> is this behaviour expected ?

No. I'll take a look into this.

**Comment by nd0ut** [comment, marked as answer]
Hey @iqbaltawakkal,

Unfortunately, I wasn't able to reproduce this issue.

Are you sure your watch callback is being executed? Could you create a minimal reproducible example (MRE) on CodeSandbox or StackBlitz?

Here is the example of how it works: https://codesandbox.io/p/devbox/uploader-minimal-vue-forked-5j7gkt?file=%2Fsrc%2FApp.vue%3A24%2C85

**Reply by iqbaltawakkal** [reply]
Hi, I just realize that my code is incorrect while looking at your example and yes it is working fine reactively.
Thank you!
